freqtrade/data/dataprovider.py

```diff
@@ -63,9 +79,99 @@ def _set_cached_df(
         :param dataframe: analyzed dataframe
         :param candle_type: Any of the enum CandleType (must match trading mode!)
         """
-        self.__cached_pairs[(pair, timeframe, candle_type)] = (
+        pair_key = (pair, timeframe, candle_type)
+        self.__cached_pairs[pair_key] = (
             dataframe, datetime.now(timezone.utc))
 
+    # For multiple producers we will want to merge the pairlists instead of overwriting
+    def set_producer_pairs(self, pairlist: List[str], producer_name: str = "default"):
+        """
+        Set the pairs received to later be used.
+        This only supports 1 Producer right now.
+
+        :param pairlist: List of pairs
+        """
+        self.__producer_pairs[producer_name] = pairlist.copy()
+
+    def get_producer_pairs(self, producer_name: str = "default") -> List[str]:
+        """
+        Get the pairs cached from the producer
+
+        :returns: List of pairs
+        """
+        return self.__producer_pairs.get(producer_name, [])
+
+    def emit_df(
+        self,
+        pair_key: PairWithTimeframe,
+        dataframe: DataFrame
+    ) -> None:
+        """
+        Send this dataframe as an ANALYZED_DF message to RPC
+
+        :param pair_key: PairWithTimeframe tuple
+        :param data: Tuple containing the DataFrame and the datetime it was cached
+        """
+        if self.__rpc:
+            self.__rpc.send_msg(
+                {
+                    'type': RPCMessageType.ANALYZED_DF,
+                    'data': {
+                        'key': pair_key,
+                        'value': dataframe
+                    }
+                }
+            )
+
+    def add_external_df(
+        self,
+        pair: str,
+        timeframe: str,
+        dataframe: DataFrame,
+        candle_type: CandleType,
+        producer_name: str = "default"
+    ) -> None:
+        """
+        Add the pair data to this class from an external source.
+
+        :param pair: pair to get the data for
+        :param timeframe: Timeframe to get data for
+        :param candle_type: Any of the enum CandleType (must match trading mode!)
+        """
+        pair_key = (pair, timeframe, candle_type)
+
+        if producer_name not in self.__producer_pairs_df:
+            self.__producer_pairs_df[producer_name] = {}
+
+        self.__producer_pairs_df[producer_name][pair_key] = (dataframe, datetime.now(timezone.utc))
+
+    def get_external_df(
```

Comment:
I don't think we need an example strategy for every detail. We can expect users to be able to copy/paste stuff from the documentation, too.

Reply:
I think at the very minimum documenting how the additional DataProvider functions work is necessary. Maybe not showing how every way to use it in a Strategy, but at least a basic understanding. The Strategy can request data on a pair from the Leader by using `dp.get_external_df`, if the data is available, then the full dataframe is returned, otherwise an empty one is returned.

Reply:
that's absolutely true - and a section in the docs must contain this segment of the strategy for sure.

Reply:
Maybe we should start to leverage the example strategy repo for stuff like this?

Reply:
not really - you're not showing a strategy - you're showing a technique. That goes into the docs.

Reply:
Well we need to show both :-)

Reply:
No need for a strategy.
A good documentation for this will suffice.